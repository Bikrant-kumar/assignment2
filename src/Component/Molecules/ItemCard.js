import React,{useState} from 'react';
import Label from '../Atoms/Label';
import AddReview from './AddReview';
import { useDispatch } from 'react-redux';
import Modal from '@material-ui/core/Modal';
import Card from '@material-ui/core/Card';
import DeleteIcon from '@material-ui/icons/Delete';
import CardMedia from '@material-ui/core/CardMedia';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CardContent from '@material-ui/core/CardContent';
import {deleteItem} from '../../Store/Action/ListAction';
import RateReviewOutlinedIcon from '@material-ui/icons/RateReviewOutlined';

const useStyles = makeStyles((theme) => ({
    root: {
      display: 'grid',
      width:"30%",
      marginTop: theme.spacing(2),
      marginLeft: "2.6%",
      borderRadius: "15px",
    },
    reviewIcon:{
      textAlign:'left'
    },
    deleteIcon:{
      marginTop:theme.spacing(-6),
      float:'right'
    }
  }));
export default function ImgMediaCard(props) {
  const classes = useStyles();
  const dispatch = useDispatch();
  const [open, setOpen] = useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  }; 

  return (
    <Card className={classes.root}>
        <CardContent >
          <CardMedia
          component="img"
          alt="Contemplative Reptile"
          height="140"
          image={props.item.Photo}
          title={props.item.Name}
          />
          <Typography gutterBottom variant="h5" component="h2" className={classes.reviewIcon}> 
            <Label text={props.item.Name}/>
          </Typography>
          <Typography className={classes.deleteIcon}>
            <IconButton  >
              {props.manager?<DeleteIcon onClick={()=>dispatch(deleteItem(props.item.id))}/>:<RateReviewOutlinedIcon onClick={handleOpen}/>}
            </IconButton>
          </Typography>
          <Typography color="textSecondary" component="p" className={classes.reviewIcon}> 
            â‚¹{props.item.Cost}
          </Typography>
          <Modal
          open={open}
          onClose={handleClose}
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          >
          <AddReview  size="small" text="ADD" title={props.item.Name} handleClose={handleClose}/> 
          </Modal>
        </CardContent>
    </Card>
  );
}