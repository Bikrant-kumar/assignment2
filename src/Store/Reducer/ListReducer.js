export const initState = {    
   
};

export function itemReducer (state = initState, action) {
    switch (action.type) {
        case "DELETE_ITEM":
            const newItem = state.item.filter((item) => item.id !== action.id);
            return{
                ...state,
                item: newItem
            };

        case "DELETE_ROW":
            const newTable= action.table.item.splice(action.table.item.indexOf(action.rowId),1)
                action.table.totalCost = action.table.totalCost - action.rowCost;
                action.table.totalItem= action.table.totalItem- 1;
            return {
                ...state,
                ...newTable
            };
        
        case "ADD_ROW":
            const newRowInTable = action.table.item.push(action.menu)
            action.table.totalCost = action.table.totalCost + action.menu.Cost;
            action.table.totalItem= action.table.totalItem + 1;
            return {
                ...state,
                ...newRowInTable
            };

        case "ADD_ITEM":
            const items = [action.item, ...state.item]
            return {
                ...state,
                item: items,
            };
        case "ADD_REVIEW":
            const review = [action.review,...state.reviews]
            return{
                ...state,
                reviews:review
            };
        
        case "GET_STATE":
            return {  
                ...action.data,
            }

        case "DELETE_ALL_ITEM":
            const newTable1 = action.table.item.splice(0,action.table.item.length)
            action.table.totalCost = 0;
            action.table.totalItem = 0;
            return {
                ...state,
                ...newTable1
            }
            
        default:
            return state;    
    }
};

 export default itemReducer;