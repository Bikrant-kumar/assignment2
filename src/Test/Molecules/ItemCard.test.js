import React from 'react';
import { screen, render } from '@testing-library/react';
import ItemCard from '../../Component/Molecules/ItemCard';
import ListReducer from '../../Store/Reducer/ListReducer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockData from '../MockData';
import { createStore } from 'redux';

describe('ItemCard',()=> {

  test('ItemCard Render',() => {

    const item=    {
        "id": "item-30",
        "Name": "Sarson Da Saag",
        "Cost": 100,
        "Serving":1,
        "Photo":"https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRZdsgv_wC5-CzghxHJuRbluRRVbkHuvHDrrA&usqp=CAU"
      };     
        render(
          <Provider store={createStore(ListReducer, MockData)}>
        <ItemCard item={item} size="small" color="primary" text="Delete"  manager="false"/>
        </Provider>);
        expect(screen.getByText("Sarson Da Saag")).toBeInTheDocument();
     })

     test('test ItemCard with snapshot',()=>{
      const item=    {
        "id": "item-30",
        "Name": "Sarson Da Saag",
        "Cost": 100,
        "Serving":1,
        "Photo":"https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRZdsgv_wC5-CzghxHJuRbluRRVbkHuvHDrrA&usqp=CAU"
      };
      const component= renderer.create(
        <Provider store={createStore(ListReducer, MockData)}>
      <ItemCard item={item} size="small" color="primary" text="Delete"  manager="true"/>
      </Provider>).toJSON();
      expect(component).toMatchSnapshot();
     })

})
 