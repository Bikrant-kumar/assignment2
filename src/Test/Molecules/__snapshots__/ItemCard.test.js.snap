// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard test ItemCard with snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <img
      alt="Contemplative Reptile"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="140"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRZdsgv_wC5-CzghxHJuRbluRRVbkHuvHDrrA&usqp=CAU"
      title="Sarson Da Saag"
    />
    <h2
      className="MuiTypography-root makeStyles-reviewIcon-6 MuiTypography-h5 MuiTypography-gutterBottom"
    >
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        Sarson Da Saag
      </div>
    </h2>
    <p
      className="MuiTypography-root makeStyles-deleteIcon-7 MuiTypography-body1"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
      </button>
    </p>
    <p
      className="MuiTypography-root makeStyles-reviewIcon-6 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      â‚¹
      100
    </p>
  </div>
</div>
`;
